x-pack/plugins/synthetics/server/routes/monitor_cruds/add_monitor.ts

```diff
@@ -75,3 +56,35 @@ export const addSyntheticsMonitorRoute: UMRestApiRouteFactory = () => ({
     return response.ok({ body: newMonitor });
   },
 });
+
+export const syncNewMonitor = async ({
+  monitor,
+  monitorSavedObject,
+  server,
+}: {
+  monitor: SyntheticsMonitor;
+  monitorSavedObject: SavedObject<EncryptedSyntheticsMonitor>;
+  server: UptimeServerSetup;
+}) => {
+  const errors = await server.syntheticsService.addConfig({
+    ...monitor,
+    id: (monitor as MonitorFields)[ConfigKey.CUSTOM_HEARTBEAT_ID] || monitorSavedObject.id,
+    fields: {
+      config_id: monitorSavedObject.id,
```

Comment:
shouldn't we push monitor source type here as well? 

or it will be set in heartbeat?

Reply:
It will be set in heartbeat. Andrew defined a new field `monitor.origin`, for it to be passed through. `ConfigKey.SOURCE_TYPE` will be mapped to that field.